src/inmanta/deploy/scheduler.py

```diff
@@ -504,6 +504,11 @@ async def _new_version(
                     added_requires[resource] = added
                 if dropped:
                     dropped_requires[resource] = dropped
+                    if self._state.resource_state[resource].blocked is BlockedStatus.TRANSIENT:
+                        dependencies = await self._get_last_non_deploying_state_for_dependencies(resource=resource)
+                        if all(dependencies[requires_id].deployed for requires_id in new_requires):
+                            self._state.resource_state[resource].blocked = BlockedStatus.NO
+                            unblocked_resources.add(resource)
```

Comment:
`unblocked_resources` has a very different meaning, we must not add these. It's an understandable mistake to make though, given its name. So perhaps we should rename it. It made sense when there was only `YES` or `NO`, but now that we have `TRANSIENT`, the name should probably indicate that it has to do with definedness.

Perhaps `newly_defined` could be a good name? It sounds a bit clunky but it's the best I can come up with. The purpose of the list is to collect resources that were previously undefined but not anymore.